### DISCUSSION TITLE
[Chapter 01]

### DISCUSSION THREAD

**Discussion by BvrPark** [cowaine/spring-boot-for-msa#2] [Chapter 01. Microservices Architecture] [answered]
1. MSA와 모놀리식 아키텍처를 실제로 사용할 때, 설계부터 궁금한 점들이 있습니다.
 - 예를 들어, 처음부터 MSA, 모놀리식 기타 등등의 아키텍처를 선정하고 설계하는지?
 - 아니면 모놀리식으로 가정하고 설계하다가 프로젝트 사이즈가 커지면 MSA로 전환하는지?
 - 만약에 MSA로 설계를 해보신 경험이 있다면 기능을 나누셨을 때, 어떤 것을 기준으로 나누셨는지?(본인만의 기준?)

2. 책 내용과는 별개로 현업에서는 요구사항이 자세한 편인지 궁금합니다.

**Comment by corock** [comment, marked as answer]
> 예를 들어, 처음부터 MSA, 모놀리식 기타 등등의 아키텍처를 선정하고 설계하는지?

어떤 시스템을 만든다고 가정하면 경험상 배포를 고려할 때, 당장 WAS 대수가 달라지기 때문에 보통 아키텍처를 고려하는 것 같습니다.

> 아니면 모놀리식으로 가정하고 설계하다가 프로젝트 사이즈가 커지면 MSA로 전환하는지?

네. 이런 경우도 현업에서 적지 않게 일어나고 있는 것으로 알고 있습니다.

> 만약에 MSA로 설계를 해보신 경험이 있다면 기능을 나누셨을 때, 어떤 것을 기준으로 나누셨는지?(본인만의 기준?)

이런 고민으로 인해 도메인이라는 개념이 나왔고, 도메인 주도 설계(_Domain-driven design_, DDD)가 나오지 않았을까 싶습니다.

> 책 내용과는 별개로 현업에서는 요구사항이 자세한 편인지 궁금합니다.

이건 정말 사바사 부바부인 것 같습니다~ 자세한 곳이 있을 수도 있구요(?)
현업에서 업무를 하다보면 요구사항이 처음부터 자세했다기 보다 커뮤니케이션을 하면서 점점 구체화되는 느낌을 자주 받습니다.
